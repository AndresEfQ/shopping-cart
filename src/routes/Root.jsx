import { useState } from "react";
import { Outlet } from "react-router-dom";
import Header from "../components/Header";
import Footer from "../components/Footer";
import Cart from "../components/Cart";
import styled from "styled-components";

export default function Root({windowWidth}) {

  const [cartIsVisible, setCartIsVisible] = useState(false);

  const toggleCart = (e) => {
    e.stopPropagation();
    setCartIsVisible((prevCartIsVisible => !prevCartIsVisible));
  }

  return (
    <Background>
      <Header toggleCart={toggleCart} windowWidth={windowWidth} />
      <div>
        <Outlet />
        {cartIsVisible && <Cart toggleCart={toggleCart} />}
      </div>
      <Footer />
    </Background>
  )
}

const Background = styled.div`
  background: no-repeat center url("https://g.foolcdn.com/editorial/images/519804/hasbro-magic-gaming-source-mtg.jpg");

  & > div {
    margin-top: 10vh;
    height: 90vh;
    width: 100%;
    display: flex;
  }
`
