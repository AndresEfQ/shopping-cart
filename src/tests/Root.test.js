import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Root from "../routes/Root";
import CartContextProvider from "../context/CartContext";

describe("Root component", () => {

  test("Root renders correctly", () => {
    const { container } = render(<CartContextProvider><Root windowWidth="1866" /></CartContextProvider>, {wrapper: BrowserRouter});
    expect(container).toMatchSnapshot();
  });

  test("Home button has home link", () => {
    render(<CartContextProvider><Root windowWidth="1866" /></CartContextProvider>, {wrapper: BrowserRouter});

    expect(screen.getByRole("link", {name: "Home"}).href).toMatch("/");
  });

  test("Shop button has shop link", () => {
    render(<CartContextProvider><Root windowWidth="1866"u /></CartContextProvider>, {wrapper: BrowserRouter});

    expect(screen.getByRole("link", {name: "Shop"}).href).toMatch("/shop");
  });
});