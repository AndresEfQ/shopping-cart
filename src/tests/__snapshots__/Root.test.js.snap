// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component Root renders correctly 1`] = `
<div>
  <div
    class="sc-jsMahE juXzZs"
  >
    <nav
      class="sc-beqWaB cPhTkj"
    >
      <a
        href="/"
      >
        <h1
          class="sc-gueYoa fWafkR"
        >
          Magic Store
        </h1>
      </a>
      <ul
        class="sc-dmqHEX jNhhdZ"
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            href="/"
          >
            Contact
          </a>
        </li>
        <li>
          <svg
            fill="currentColor"
            height="28"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
            />
          </svg>
        </li>
      </ul>
    </nav>
    <div />
    <footer
      class="sc-jTrPJq ljlZtm"
    >
      Copyright Â© 
      <a
        class="sc-gLDzan dIEA-DD"
        href="https://github.com/AndresEfQ/shopping-cart"
      >
        AndresEfQ 2023 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 496 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          />
        </svg>
      </a>
    </footer>
  </div>
</div>
`;
